docs/proposals/affinity.md

```diff
@@ -0,0 +1,341 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/affinity.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Inter-pod topological affinity and anti-affinity
+
+## Introduction
+
+This document describes a proposal for specifying and implementing inter-pod topological affinity and
+anti-affinity. By that we mean: rules that specify that certain pods should be placed
+in the same topological domain (e.g. same node, same rack, same zone, same
+power domain, etc.) as some other pods, or, conversely, should *not* be placed in the
+same topological domain as some other pods.
+
+Here are a few example rules; we explain how to express them using the API described
+in this doc later, in the section "Examples."
+* Affinity
+  * Co-locate the pods from a particular Service or Job in the same availability zone
+  * When scheduling a pod that uses a license for a particular piece of software, try
```

Comment:
IMHO you would need dynamic labeling (a.k.a - attributes, or machine_ad) for persistence in the windows license.  But you may also want it for hot-tracking of images or data, etc.